src/supervisor/Supervisor.h

```diff
@@ -233,15 +204,16 @@ class Supervisor : public iosource::IOSource {
          */
         std::vector<std::string> addl_user_scripts;
         /**
-         * Environment variables and values  to define in the node.
+         * Environment variables and values to define in the node.
          */
         std::map<std::string, std::string> env;
         /**
-         * The Cluster Layout definition.  Each node in the Cluster Framework
+         * The Cluster layout definition.  Each node in the Cluster Framework
          * knows about the full, static cluster topology to which it belongs.
-         * Entries in the map use node names for keys.
+         * Entries in the map use node names for keys and the JSON representations
+         * of script-layer Supervisor::ClusterEndpoint records as values.
          */
-        std::map<std::string, ClusterEndpoint> cluster;
+        std::map<std::string, std::string> cluster;
```

Comment:
Only thought is whether we could just keep the whole JSON blob in a `std::string` - is there any advantage to having the `std::map` ?

Reply:
Thanks Arne. There was no reason other than focusing the change on `ClusterEndpoint` instances, but I think you're right — with that change the map itself isn't required. I'll make an edit pass, also for your other suggestions.